### PR TITLE
fix bug when syncing distribution without schema and remove unused logic

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #23085

### CODE REVIEW

**Comment by MrPresent-Han** [comment]
rerun ut

File: internal/querynodev2/services.go [hunk 1]
```diff
@@ -1011,10 +999,18 @@ func (node *QueryNode) SyncDistribution(ctx context.Context, req *querypb.SyncDi
...
+				commonpbutil.WithMsgID(req.Base.GetMsgID()),
+			),
 			Infos:        infos,
+			Schema:       req.GetSchema(),
+			LoadMeta:     req.GetLoadMeta(),
 			CollectionID: req.GetCollectionID(),
+			ReplicaID:    req.GetReplicaID(),
+			DstNodeID:    nodeID,
+			Version:      time.Now().UnixNano(),
+			NeedTransfer: false,
```
Comment (congqixia) [inline internal/querynodev2/services.go:1013]: `Version` shall be `qc` allocated value.

Comment (MrPresent-Han) [reply internal/querynodev2/services.go:1013]: yes, that's truly a defect, I will change this

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**